R/derive_joined.R

```diff
@@ -443,3 +541,362 @@ derive_vars_joined <- function(dataset,
+#'   considered, i.e., the observation is not flagged.
+#'
+#'   This parameter should be specified if `filter_join` contains summary
+#'   functions which should not apply to all observations but only from a
+#'   certain observation before the current observation up to the current
+#'   observation.
+#'
+#' @param first_cond_upper Condition for selecting range of data (after)
+#'
+#'   If this argument is specified, the other observations are restricted up to
+#'   the first observation where the specified condition is fulfilled. If the
+#'   condition is not fulfilled for any of the other observations, no
+#'   observations are considered, i.e., the observation is not flagged.
+#'
+#'   This parameter should be specified if `filter_join` contains summary
+#'   functions which should not apply to all observations but only up to the
+#'   confirmation assessment.
+#'
+#' @param filter_join Filter for the joined dataset
+#'
+#'   The specified condition is applied to the joined dataset. Therefore
+#'   variables from both datasets `dataset` and `dataset_add` can be used.
+#'
+#'   Variables created by `order` or `new_vars` arguments can be used in the
+#'   condition.
+#'
+#'   The condition can include summary functions like `all()` or `any()`. The
+#'   joined dataset is grouped by the original observations.
+#'
+#'   *Permitted Values*: a condition
+#'
+#' @param mode Selection mode
+#'
+#'   Determines if the first or last observation is selected. If the `order`
+#'   argument is specified, `mode` must be non-null.
+#'
+#'   If the `order` argument is not specified, the `mode` argument is ignored.
+#'
+#'   *Permitted Values*: `"first"`, `"last"`, `NULL`
+#'
+#' @param check_type Check uniqueness?
+#'
+#'   If `"warning"` or `"error"` is specified, the specified message is issued
+#'   if the observations of the (restricted) joined dataset are not unique with
+#'   respect to the by variables and the order.
+#'
+#'   This argument is ignored if `order` is not specified. In this case an error
+#'   is issued independent of `check_type` if the restricted joined dataset
+#'   contains more than one observation for any of the observations of the input
+#'   dataset.
+#'
+#'   *Permitted Values*: `"none"`, `"warning"`, `"error"`
+#'
+#'
+#' @details
+#'
+#' 1. The variables specified by `order` are added to the additional dataset
+#' (`dataset_add`).
+#'
+#' 1. The variables specified by `join_vars` are added to the additional dataset
+#' (`dataset_add`).
+#'
+#' 1. The records from the additional dataset (`dataset_add`) are restricted to
+#' those matching the `filter_add` condition.
+#'
+#' 1. The input dataset and the (restricted) additional dataset are left joined
+#' by the grouping variables (`by_vars`). If no grouping variables are
+#' specified, a full join is performed.
+#'
+#' 1. The joined dataset is restricted by the `filter_join` condition.
+#'
+#' @keywords internal
```

Comment:
```suggestion
#' @keywords internal
#' @export
```

Although it will only be used internally, are we not exporting this function at all?

Reply:
I think if we export it, it should be included on the "Reference" page, i.e., it should not be internal.

Do you think it users would like to use it?

Reply:
I doubt our users would use it, but it feels like it has similar "use" as `assert_terms()` and `assert_valid_queries()` which are both exported, more for a consistency thing

Reply:
Yes, but it seems that we want to change this (see https://github.com/pharmaverse/admiral/discussions/2085#discussioncomment-7351305).
@bms63 , what do you think?